.github/actions/deploy-frontend/action.yml

```diff
@@ -0,0 +1,41 @@
+# action.yml
+name: 'Upload Static Site (React)'
+description: 'Push react website to Azure blob storage'
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: Set environment variables - dev
+      if: github.ref == 'refs/heads/dev-rheft'
+      shell: bash
+      run: echo "IMAGE_REPO=rkh5012" >> $GITHUB_ENV
+
+    - name: Set environment variables - test
+      if: github.ref == 'refs/heads/ronheft/test-investigation'
+      shell: bash
+      run: echo "IMAGE_REPO=pdhtest" >> $GITHUB_ENV
+
+    - name: Set environment variables - staging
+      if: github.ref == 'refs/heads/master'
+      shell: bash
+      run: echo "IMAGE_REPO=pdhstaging" >> $GITHUB_ENV
+
+    - name: Set environment variables - production
+      if: github.ref == 'refs/heads/production'
+      shell: bash
+      run: echo "IMAGE_REPO=pdhprod" >> $GITHUB_ENV
+
+    - name: Download Frontend Artifact (React)
+      uses: actions/download-artifact@v2
+      with:
+        name: static_website_react-${{ github.run_id }}
+        path: frontend-react/build
+
+    - name: Upload Static Site (React)
+      if: env.IMAGE_REPO
```

Comment:
Only upload if env.IMAGE_REPO exists.
Currently no error message, just does not deploy. Can be improved.